### ISSUE TITLE
Popup state

### ISSUE THREAD

**Issue by drwlrsn** [gabesmed/ember-leaflet#77]
I was wondering if anyone had any ideas on how to reflect the popup state in the URL.
For example, if I had a map with a bunch of POIs and a user clicked the marker for a particular market, it would be great if the URL would change something like `/supermarkets/3424` or `/supermarket/<some controller property>`.

I have been digging around descending `SomeCollectionLayer._childLayers[n]._childLayers[n]`. This seems like the really wrong way to do this. How should I approach this problem?

**Comment by codefox421** [comment]
What if popups were componentized as compositional components:

```
{{!-- template.hbs --}}
...
  {{#map-component}}
    {{tile-layer}}
    {{#marker-layer}}
      {{#marker-item}}
        {{#marker-popup}}
          <p>Popup content as HTML, with {{bindableProperties}}.</p>
        {{/marker-popup}}
      {{/marker-item}}
    {{/marker-layer}}
  {{/map-component}}
...
```

Or something like that. (Forgive me, I didn't think very long about the entire component hierarchy.) So to reflect popup content in the URL you would do:

```
{{#marker-popup}}
  {{outlet}}
{{/marker-popup}}
```

And let Ember's router take care of the rest by using nested routes.

**Comment by miguelcobain** [comment]
Related: https://github.com/gabesmed/ember-leaflet/pull/98#issuecomment-78551161

@codefox421 something like that is planned for future versions.

**Comment by codefox421** [comment]
Yep, I remember that. Basically, it solves this case with very little "heavy lifting" on EmberLeaflet's side. Plus it looks a lot cleaner and is generally more flexible than something like `{{popup-marker-layer popupContent=someComputedProp}}`.

This is just a thing to keep in mind when implementing those components.
